api/src/main/java/org/apache/iceberg/actions/RewritePositionDeleteFiles.java

```diff
@@ -41,11 +119,60 @@
   RewritePositionDeleteFiles filter(Expression expression);
 
   /** The action result that contains a summary of the execution. */
+  @Value.Immutable
   interface Result {
+    List<PositionDeleteGroupRewriteResult> rewriteResults();
+
     /** Returns the count of the position deletes that been rewritten. */
     int rewrittenDeleteFilesCount();
```

Comment:
Shall we add default implementation to aggregate the results from individual results?

Reply:
Good idea, done